be/src/storage/rowset/segment_v2/array_column_iterator.cpp

```diff
@@ -122,6 +122,64 @@ Status ArrayColumnIterator::next_batch(size_t* n, vectorized::Column* dst) {
     return Status::OK();
 }
 
+Status ArrayColumnIterator::next_batch(vectorized::SparseRange& range, vectorized::Column* dst) {
+    vectorized::ArrayColumn* array_column = nullptr;
+    vectorized::NullColumn* null_column = nullptr;
+    if (dst->is_nullable()) {
+        auto* nullable_column = down_cast<vectorized::NullableColumn*>(dst);
+
+        array_column = down_cast<vectorized::ArrayColumn*>(nullable_column->data_column().get());
+        null_column = down_cast<vectorized::NullColumn*>(nullable_column->null_column().get());
+    } else {
+        array_column = down_cast<vectorized::ArrayColumn*>(dst);
+    }
+
+    // 1. Read null column
+    if (_null_iterator != nullptr) {
+        RETURN_IF_ERROR(_null_iterator->next_batch(range, null_column));
```

Comment:
Check whether `null_column` is nullptr before calling `next_batch`

Reply:
done